File: Lab4_Java/src/Company.java
import java.util.Scanner;

public class Company {
    private String brand;
    private Devices devices;
    private Creation creation;
    private int n_sp = 0;
    private int n_lt = 0;
    public Company(String brand, Devices devices, Creation creation)
    {
        this.brand = brand;
        this.devices = devices;
        this.creation = creation;
    }
    public Company()
    {
        this.devices = new Devices();
        this.creation = new Creation();
        this.brand = "Brand";
    }
    public int getN_sp()
    {
        return n_sp;
    }
    public int getN_lp()
    {
        return n_lt;
    }
    public void input()
    {
        Scanner inp = new Scanner(System.in);
        System.out.printf("Введите название компании ");
        brand = inp.next();
        System.out.printf("Введите число моделей смартфонов вашей компании ");
        n_sp = inp.nextInt();
        System.out.printf("Введите число моделей ноутбуков вашей компании ");
        n_lt = inp.nextInt();
        devices.input_dev(n_sp, n_lt);
        creation.input_creat();
    }
    public void output(int n_sp, int n_lt)
    {
        System.out.printf("Название компании: %s\n", brand);
        this.devices.output_dev(n_sp, n_lt);
        this.creation.output_creat();
    }
    public void outputstr(int n_sp, int n_lt)
    {
        devices.outstr_dev(n_sp, n_lt);
    }
    public float income(int n_sp, int n_lt)
    {
        float income, taxes;
        taxes = creation.get_taxes();
        income = (float)devices.income_dev(n_sp, n_lt) * (100 - taxes) / 100;
        return income;
    }
    public int expenses()
    {
        return creation.expenses_creat();
    }
    public float profit(int n_sp, int n_lt)
    {
        return income(n_sp, n_lt) - expenses();
    }
}

File: Lab4_C++/company.cpp
#include "company.h"
#include <iostream>
#include <string>
using namespace std;

Company::Company(const Company& company)
{
    brand = string(company.brand);
    devices = Devices(company.devices);
    creation = Creation(company.creation);
}

Company::Company(string brand, Devices devices, Creation creation)
{
    this->brand = brand;
    this->devices = devices;
    this->creation = creation;
}

Company::Company()
{
    devices = Devices();
    creation = Creation();
    brand = "Brand";
}

Company::Company(string brand)
{
    this->brand = brand;
    devices = Devices();
    creation = Creation();
}

Company::~Company() {}

void Company::input(int* n_sp, int* n_lt)
{
    cout << "Введите название компании" << endl;
    cin >> brand;
    cout << "Введите число моделей смартфонов вашей компании" << endl;
    cin >> *n_sp;
    cout << "Введите число моделей ноутбуков вашей компании" << endl;
    cin >> *n_lt;
    devices.input_dev(n_sp, n_lt);
    creation.input_creat();
}

void Company::output(int n_sp, int n_lt)
{
    cout << brand << endl;
    devices.output_dev(n_sp, n_lt);
    creation.output_creat();
}

void Company::outputstr(int n_sp, int n_lt)
{
    devices.outstr_dev(n_sp, n_lt);
}


float Company::income(int n_sp, int n_lt)
{
    float income;
    income = devices.income_dev(n_sp, n_lt) * (100 - creation.taxes) / 100;
    return income;
}

int Company::expenses()
{
    return creation.expenses_creat();
}

int Company::profit(int n_sp, int n_lt)
{
    return income(n_sp, n_lt) - expenses();
}